bin/check-alertclear-patterns.rb

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env ruby
+#
+# Check Alert/Clear Patterns plugin
+# ===
+# This plugin will read a log file and raise an alert on each line matching
+# #{pattern_alert}. This same alert will be cleared only when a new line
+# is found by matching #{pattern_clear}. Alert/Clear relationship is based on
+# a subpart of both lines, using #{pattern_identifier} regexp to extract the id
+#
+# This plugin checks a log file, skipping lines that have already been read,
+# like Nagios's check_log. However, instead of making a backup copy of the
+# whole log file (very slow with large logs), it stores the number of bytes
+# read, and seeks to that position next time.
+#
+# ===
+# Example:
+# Log file content:
+# 16:00: lost connection to database PRIMARY
+# 16:01: lost connection to database SECONDARY
+# 16:20: connected to database PRIMARY
+#
+# Options will be:
+# -A "lost connection"
+# -C "connected to"
+# -I "database (.*)"
+#
+# Sensu events generated will be:
+# {"check": {"name":"PRIMARY",  "status": 2, "output": "CRITICAL: 16:00: lost connection to database PRI01"} }
+# {"check": {"name":"SECONDARY","status": 2, "output": "CRITICAL: 16:01: lost connection to database PRI02"} }
+# {"check": {"name":"PRIMARY",  "status": 0, "output": "OK: 16:20: connected to database PRI01"} }
+#
+# ===
+# Copyright 2014 Thomson Reuters, Jonathan Huot <jonathan.huot@thomsonreuters.com>
+# Copyright 2011 Sonian, Inc <chefs@sonian.net>
+#
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+# for details.
+
+require 'rubygems' if RUBY_VERSION < '1.9.0'
+require 'sensu-plugin/check/cli'
+require 'fileutils'
+require 'json'
+require 'socket'
+require 'time'
+
+class CheckLog < Sensu::Plugin::Check::CLI
+  BASE_DIR = '/var/cache/check-log'.freeze
```

Comment:
Make this an option and avoid constants for things that could change (locally I don't have `/var/cache/check-log` and this makes me rather than specify a dir must create it on my local box for testing which is not a great ux.